frontend/scripts/check-patternfly-modules.sh

```diff
@@ -8,50 +8,68 @@ COL_YELLOW='\033[33m'
 
 resolution_errors=false
 
-# Ensure that only one version resolution exists for PatternFly packages in yarn.lock file.
+# Ensure that only one major version resolution exists for the given package in yarn.lock file.
 # Having multiple version resolutions may lead to bugs and webpack build or performance issues.
-#
-# Note: this code relies on yarn.lock v1 format since `yarn why <query>` CLI does not support
-# proper JSON output. We should revisit this code once we upgrade to a newer Yarn version.
 check-resolution() {
   local PKG_NAME="${1:?Provide package name to check}"
-  local RES_COUNT=$(grep -c "^\"${PKG_NAME}@" yarn.lock)
+  local PKG_MAJOR_VERSION="${2:?Provide package major version}"
+
+  # https://stackoverflow.com/a/29613573
+  local PKG_NAME_ESCAPED=$(sed 's/[^^]/[&]/g; s/\^/\\^/g' <<< "${PKG_NAME}")
+
+  local RES_COUNT=$(
+    yarn why "${PKG_NAME}" --json 2>/dev/null \
+      | grep -o '=> Found \\".*\\"' \
+      | sed -n "s/^.*${PKG_NAME_ESCAPED}@\(.*\)[\]\"/\1/p" \
+      | grep "^${PKG_MAJOR_VERSION}\." \
+      | uniq \
+      | wc -l
+  )
+
+  local MSG_PKG_ERROR="${COL_RED}${PKG_NAME}${COL_RESET}"
+  local MSG_PKG_SUCCESS="${COL_GREEN}${PKG_NAME}${COL_RESET}"
+  local MSG_VERSION="${COL_YELLOW}${PKG_MAJOR_VERSION}.x${COL_RESET}"
 
   if [[ $RES_COUNT -eq 0 ]]; then
-    echo -e "${COL_RED}${PKG_NAME}${COL_RESET} has no version resolutions"
+    echo -e "${MSG_PKG_ERROR} has no ${MSG_VERSION} resolutions"
     resolution_errors=true
   elif [[ $RES_COUNT -ne 1 ]]; then
-    echo -e "${COL_RED}${PKG_NAME}${COL_RESET} has multiple version resolutions"
+    echo -e "${MSG_PKG_ERROR} has multiple ${MSG_VERSION} resolutions"
     resolution_errors=true
+  else
+    echo -e "${MSG_PKG_SUCCESS} has one ${MSG_VERSION} resolution"
   fi
 }
 
-# These packages will be checked for version resolutions.
-# This list is explicit since we may want to skip checking certain packages.
-#
-# To see the current full list, run the following command:
-#   sed -nr 's/^\"(@patternfly\/[^@]+).*$/\1/p' yarn.lock | sort -u
+# List of packages to check, formatted as '{package_name}:{major_version}'
 PKGS_TO_CHECK=(
-  '@patternfly/quickstarts'
-  '@patternfly/react-catalog-view-extension'
-  '@patternfly/react-charts'
-  '@patternfly/react-component-groups'
-  '@patternfly/react-console'
-  '@patternfly/react-core'
-  '@patternfly/react-icons'
-  '@patternfly/react-log-viewer'
-  '@patternfly/react-styles'
-  '@patternfly/react-table'
-  '@patternfly/react-tokens'
-  '@patternfly/react-topology'
-  '@patternfly/react-user-feedback'
-  '@patternfly/react-virtualized-extension'
+  '@patternfly/patternfly:5'
+  '@patternfly/patternfly:4'
+  '@patternfly/quickstarts:5'
```

Comment:
for `quickstarts` we have [both version 4 and 5](https://github.com/openshift/console/blob/master/frontend/package.json#L142)... even though the version 4 is pointing to `patternfly/quickstarts@2.4.2`

Reply:
@jhadvig `yarn why @patternfly/quickstarts` shows just one resolution
```
=> Found "@patternfly/quickstarts@5.1.0"
```
Our `@patternfly-4/*` aliased packages are excluded from this list - it only contains unaliased PatternFly packages that may appear in multiple versions due to some of them depending on other versions.

A good example of this is `@patternfly/patternfly` package - see `yarn why @patternfly/patternfly` resolutions
```
=> Found "@patternfly/patternfly@5.2.1"
=> Found "@patternfly-4/react-console#@patternfly/patternfly@4.139.2"
```
Notice the `@patternfly/patternfly@4.139.2` resolution that is pulled in transitively via `@patternfly-4/react-console` aliased package.

Reply:
Thanks for explanation @vojtechszocs 👍